src/libpriv/rpmostree-kernel.cxx

```diff
@@ -416,18 +416,22 @@ rpmostree_finalize_kernel (int rootfs_dfd, const char *bootdir, const char *kver
       if (contents == NULL)
         return FALSE;
 
-      /* rather than trying to parse and understand the *sum format, just hackily replace */
-      g_autofree char *old_path = g_strconcat ("  /boot/vmlinuz-", kver, NULL);
-      g_autofree char *new_path = g_strconcat ("  vmlinuz-", kver, NULL);
-      g_autofree char *new_contents = rpmostree_str_replace (contents, old_path, new_path, error);
+      /* hack: fallback for when the shasum file contains a full path */
+
+      g_auto (GStrv) split_contents = g_strsplit (contents, "  ", 2);
+      if (g_strv_length (split_contents) != 2)
+        return glnx_throw (error, "Missing filename in .vmlinuz.hmac: %s", contents);
+
+      g_auto (GStrv) path_components = g_strsplit (split_contents[1], "/", -1);
+      guint path_components_length = g_strv_length (path_components);
+      if (strlen (path_components[path_components_length - 1]) == 0)
+        return glnx_throw (error, "Path in .vmlinuz.hmac is a directory: %s", contents);
```

Comment:
How about instead of this, we do something like:

```c
if (strchr (split_contents[1], '/') != 0)
  {
    const char *basename = glnx_basename (split_contents[1]);
    // g_strconcat() && glnx_file_replace_contents_at() here
  }
```

?

I.e. use `glnx_basename` instead of splitting the path, and avoid updating the file at all which I agree with @lucab would be good to do.